src/components/App.tsx

```diff
@@ -42,14 +47,58 @@ const App: FC<AppProps> = ({ demoId }) => {
     setEvaluatePath(true);
   }, []);
 
+  const setMcodeRecords = useCallback((resources: DomainResource[]): void => {
+    // Create a Bundle for the CQL engine
+    const bundle = {
+      resourceType: 'Bundle',
+      type: 'searchset',
+      entry: resources.map((r: DomainResource) => ({ resource: r }))
+    };
+    getFixture('mCODE.cql')
+      .then(cql => convertBasicCQL(cql))
+      .then(elm => {
+        let elmResults = executeElm(bundle, elm);
+        const patientIds = Object.keys(elmResults.patientResults);
+        const mcodeData = elmResults.patientResults[patientIds[0]];
+
+        const mcodeElements: McodeElements = {
+          primaryCancer: mcodeData['Primary Cancer Condition Code'][0] ?? undefined,
+          laterality: mcodeData['Primary Cancer Condition Body Location Code'][0] ?? undefined,
+          tumorCategory:
+            mcodeData['TNM Clinical Primary Tumor Category Data Value (T Category)'][0] ??
```

Comment:
To make this a little more robust, instead of just fetching the `text` field in the referenced CQL could these also look at, for example, the `code.display` field? We don't have to account for all the many possibilities here, like interpreting codes if both text and code.display are empty (which is totally legal), but test data may not always include the text field. Maybe it's a helper function in the CQL or in the code here but it shouldn't be too complex

Reply:
That's a good point @dehall.  From my experience, and without any evidence to back it up, display is more common (and doesn't surprise me because it is taken right from the system).

Reply:
I'll update it to read `code[0].coding[0].display.value`